SCClassLibrary/JITLib/ProxySpace/NodeProxy.sc

```diff
@@ -896,19 +896,26 @@ NodeProxy : BusPlug {
 	// allocation
 
 	freeBus {
-		var oldBus = bus, c;
+		var oldBus = bus;
 		if(oldBus.isNil) { ^this };
-		if(this.isPlaying) {
-			c = (clock ? TempoClock.default);
-			c.sched(server.latency ? 0.01 + quant.nextTimeOnGrid(c) + this.fadeTime, { oldBus.free(true); nil });
-			CmdPeriod.doOnce { if(oldBus.index.notNil) { oldBus.free(true) } };
-		} {
-			oldBus.free(true)
-		};
+		this.schedAfterFade({
+			if(oldBus.index.notNil) { oldBus.free(true) }
+		});
 		busArg = bus = nil;
 		busLoaded = false;
 	}
 
+	schedAfterFade { |func|
+		var delta, usedClock;
+		if(this.isPlaying, {
+			usedClock = clock ? TempoClock.default;
+			delta = server.latency ? 0.01 + this.fadeTime; // 0.01 is reasonable for a local server
+			delta = delta + (quant ? 0).nextTimeOnGrid(usedClock);
+			usedClock.schedAbs(delta, { func.value; func = nil });
```

Comment:
Being fussy, but why not like this? `delta`'s handling was more complex in an earlier version of this PR, but now it's simpler. Also, for legibility, this is `schedAfterFade` so perhaps move `fadeTime` to the front.

```
usedClock.schedAbs(
	this.fadeTime + (server.latency ? 0.01) + (quant ? 0).nextTimeOnGrid(usedClock),
	{ func.value; func = nil }
);
```